### PR TITLE
Move some shared logic in module utils/sns

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Move some shared logic in module utils/sns

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sns_topic
sns
sns_topic_info

### CODE REVIEW

**Review by tremble — CHANGES_REQUESTED** [review]
I'm running a slightly stricter set of pylint tests locally that I think are worth fixing:
```
ERROR: plugins/module_utils/sns.py:125:0: missing-final-newline: Final newline missing
ERROR: plugins/modules/sns_topic.py:287:0: unused-import: Unused import re
ERROR: plugins/modules/sns_topic.py:288:0: unused-import: Unused import copy
```

**Comment by tremble** [comment]
(but otherwise looks good)
Thanks for this @alinabuzachis.

**Review by tremble — APPROVED** [review]

File: plugins/module_utils/sns.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+@AWSRetry.jittered_backoff(catch_extra_error_codes=['NotFound'])
+def _list_topic_subscriptions_with_backoff(client, topic_arn):
+    paginator = client.get_paginator('list_subscriptions_by_topic')
+    return paginator.paginate(TopicArn=topic_arn).build_full_result()['Subscriptions']
+
+
+@AWSRetry.jittered_backoff(catch_extra_error_codes=['NotFound'])
+def _list_subscriptions_with_backoff(client):
+    paginator = client.get_paginator('list_subscriptions')
+    return paginator.paginate().build_full_result()['Subscriptions']
```
Comment (marknet15) [inline plugins/module_utils/sns.py:32]: Is it worth combining these in a single function with a param input? for example:
https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/cloudfront_info.py#L319

Might make it easier then if any different paginators get used later

**Review by markuman — APPROVED** [review]

**Comment by markuman** [comment]
recheck

**Pull request merged**